### ISSUE TITLE
Can’t seem to get correct voltage on osd.  

### ISSUE THREAD

**Issue by droplow77** [ramiss/arduino_DJI_03_RC_ARM#28]
I tried using resistor values for the voltage regulator as listed in the readme file, and the osd shows approx 2.1v (fluctuating up and down) for a 4S battery measuring 15.8v.  

I tried changing the value of VOLT_DIVIDER up and down and I get the voltage closer, but, never exact.  It also still fluctuates up and down about 2-3v, which isn’t useful.  

Any help is greatly appreciated.  

Thank You for this amazing project.

**Comment by droplow77** [comment]
Also getting error “incompatible goggles, air unit, receiver” message in the goggles after about 30 seconds being powered up

**Comment by ramiss** [comment]
@droplow77 - For the voltage - it is not so important that the resistors are exact, but much more important that the ratio between the values is close to what is listed in the readme file.  It is also possible that your resistors are not of high quality and you might change them.

For the incompatibility, this is due to the goggles and air unit not being on the same firmware versions. Make sure both are on the latest.  Re-flash the latest even if you think it is there.

**Comment by droplow77** [comment]
@ramiss 
Thanks for the reply.  

I used the exact values (ordered on Amazon) that you listed.  Possibly low quality.  ¯\_(ツ)_/¯ I also tried different resistors previously, before my Amazon resistors arrived, by combining ones I had on hand in series/parallel to create the proper resistance value. 

If I was successful at getting the values within the correct ratio, do I need to adjust the value of VOLT_DIVIDER to get the osd voltage to correctly match the lipo?

In your testing, does your osd voltage fluctuate up and down within a 2v range, like mine? Or is it constant and holding?

**Comment by ramiss** [comment]
If you are getting fluctuations without load then something fishy is going on. This is the most basic way to sense voltage and not perfect, but you should be able to keep a stable value.  If you really want accuracy, you should pull your cruise throttle load and then calibrate the volt_divider value to actuals.

**Comment by droplow77** [comment]
Yes, I am getting fluctuating value in the osd without load.   It is fluctuating between .7 and 2.1 volts, approximately.  

What should my approximate voltage reading be at the A0 pad be if I have a charged 4S battery?   

I assume, based on your calculations, 16.8 / 6.471 = 2.59, but, I want to verify my assumption, so I can troubleshoot my voltage divider.  

I assume it needs to be within the range of 0-3.3v???


Thanks again for your help.

**Comment by ramiss** [comment]
You can use a voltage divider calculator like this:
[https://www.digikey.com/en/resources/conversion-calculators/conversion-calculator-voltage-divider](https://www.digikey.com/en/resources/conversion-calculators/conversion-calculator-voltage-divider)

I've found them to be relatively accurate.  One thought is that you somehow did go over 3.3 volts and damaged that pin.  You might try another pin and yes, keep your max potential voltage under 3.3.

**Comment by droplow77** [comment]
Thanks for your answer. 
I checked my voltage regulator with a multimeter and it is showing a stable voltage just above 2v with a half charged 4S. 

I will try another pin next.  Thanks for the advice and clarification.

**Comment by droplow77** [comment]
I think I damaged my A0 pin as you suggested.   

Using A1, I got more stable voltage, only fluctuating by approximately 0.4v. 

Edit: it is actually fluctuating in a range of .9v up and down, still. 


Thank You for your input.  

Now, I need to adjust my VOLT_DIVIDER  value.  

edit:
I found the value of VOLT_DIVIDER that most closely provided the supply voltage on the osd , but, when I plugged in a slightly lower voltage battery to test it as the battery drained in flight, the osd voltage was way off.  

Not sure how to fix this.  I think I will have to use receiver telemetry to get my voltage.  :/

Thanks again for your help.
